WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsViewController.swift

```diff
@@ -47,43 +47,49 @@ struct InPersonPaymentsView: View {
                         ServiceLocator.analytics.track(.cardPresentPaymentGatewaySelected, withProperties: ["payment_gateway": plugin.pluginName])
                     }
                 } else if viewModel.userIsAdministrator {
-                    InPersonPaymentsPluginConflictAdmin(onRefresh: viewModel.refresh)
+                    InPersonPaymentsPluginConflictAdmin(analyticReason: viewModel.state.reasonForAnalytics, onRefresh: viewModel.refresh)
```

Comment:
> As we're using viewModel.state.reasonForAnalytics multiple times and clutters the View, should we use a computed variable instead?

Yeah this seems like a better way... it's certainly less cluttered.

> That said, would it make sense to remove these entirely as a parameter from these Views, but instead add it directly to the InPersonPaymentsViewModel when is initialized?

No, not really. The viewModel has a long-lived useCase which will produce multiple `state` values over its lifetime. 

That means we shouldn't really make this a property of the VM, in my opinion. It's a property of the state, so we can't make it lazy or static/class bound on the viewModel. 

I think the cleanest way to resolve this is to make `reasonForAnalytics` non-optional on the `CardPresentPaymentOnboardingState` where it's defined. The only `nil` states currently are `loading` and `completed`, which we don't currently log the `card_present_payments_onboarding_not_complete` analytics for, so we would need to also add another check to make sure we didn't start tracking those.

I would provide values for those because I could imagine us logging other analytics for those states in future.

Reply:
However, to do this well I need to remove the `inPersonPaymentGatewaySelection` feature flag, which is fully enabled now, so I'm going to do it in a separate PR :)